File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/network/core/HttpResponse.java


package com.example.javademogithubpractice.network.core;



import androidx.annotation.Nullable;

import retrofit2.Response;



public class HttpResponse <T extends Object> {

    private Response<T> oriResponse;

    public HttpResponse(Response<T> response){
        oriResponse = response;
    }

    public boolean isSuccessful(){
        return oriResponse.isSuccessful();
    }

    public boolean isFromCache(){
        return  isResponseEnable(oriResponse.raw().cacheResponse());
    }

    public boolean isFromNetWork(){
        return  isResponseEnable(oriResponse.raw().networkResponse());
    }

    private boolean isResponseEnable(@Nullable okhttp3.Response response){
        return response != null && response.code() == 200;
    }

    public Response<T> getOriResponse() {
        return oriResponse;
    }

    public T body(){
        return oriResponse.body();
    }
}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/http/core/HttpResponse.kt
package com.example.kotlindemogithubproject.http.core

import androidx.annotation.Nullable
import retrofit2.Response


class HttpResponse<T : Any>(val oriResponse: Response<T>) {

    val isSuccessful: Boolean
        get() = oriResponse.isSuccessful

    val isFromCache: Boolean
        get() = isResponseEnable(oriResponse.raw().cacheResponse())

    val isFromNetWork: Boolean
        get() = isResponseEnable(oriResponse.raw().networkResponse())

    private fun isResponseEnable(@Nullable response: okhttp3.Response?): Boolean {
        return response != null && response.code() == 200
    }

    fun body(): T? {
        return oriResponse.body()
    }
}
